lib/strong_migrations/migration.rb

```diff
@@ -186,6 +195,24 @@ def backfill_code(table, column, default)
       end
     end
 
+    def foreign_key_code(from_table, to_table)
+      if ActiveRecord::VERSION::MAJOR > 5 || ActiveRecord::VERSION::MAJOR == 5 && ActiveRecord::VERSION::MINOR >= 2
+        <<~CODE.strip
+          #{command_str(:add_foreign_key, [from_table, to_table, validate: false])}
+              #{command_str(:validate_foreign_key, [from_table, to_table])}
+        CODE
+      else
+        <<~CODE.strip
+          safety_assured {
+                execute <<~SQL
+                  ALTER TABLE #{from_table} ADD CONSTRAINT fk_#{from_table}_#{to_table} FOREIGN KEY (#{to_table.to_s.singularize}_id) REFERENCES #{to_table} (id) NOT VALID;
+                  ALTER TABLE #{from_table} VALIDATE CONSTRAINT fk_#{from_table}_#{to_table};
```

Comment:
While reflecting on this solution, I realized that it probably isn't necessary to output the solution for old versions of Rails, because this problem itself cannot be produced on old versions since the `:validate` option didn't exist. Do we agree on that?

Reply:
Older versions of Rails behave as `validate: true`, which is unsafe, so think this is needed. Also need a down migration.